libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -0,0 +1,234 @@
+const ServiceSelection = require('../../service-selection/ServiceSelection')
+const { timeRequestsAndSortByVersion } = require('../../utils/network')
+const { CREATOR_NODE_SERVICE_NAME, DECISION_TREE_STATE } = require('./constants')
+
+class CreatorNodeSelection extends ServiceSelection {
+  constructor ({ creatorNode, numberOfNodes, ethContracts, whitelist, blacklist }) {
+    super({
+      getServices: async () => {
+        this.currentVersion = await ethContracts.getCurrentVersion(CREATOR_NODE_SERVICE_NAME)
+        const services = await this.ethContracts.getServiceProviderList(CREATOR_NODE_SERVICE_NAME)
+        return services.map(e => e.endpoint)
+      },
+      whitelist,
+      blacklist
+    })
+    this.creatorNode = creatorNode
+    this.numberOfNodes = numberOfNodes
+    this.ethContracts = ethContracts
+    // Services with the structure {request, response, millis} (see timeRequest) sorted by semver and response time
+    this.healthCheckedServicesList = []
+    this.healthCheckPath = 'version'
+  }
+
+  /**
+   * Selects a primary and secondary creator nodes. Order of preference is highest version, then response time.
+   *
+   * 1. Retrieve all the creator node services
+   * 2. Filter from/out creator nodes based off of the whitelist and blacklist
+   * 3. Filter out unhealthy, outdated, and still syncing nodes via health and sync check
+   * 4. Sort by healthiest (highest version -> lowest version); secondary check if equal version based off of responseTime
+   * 5. Select a primary and numberOfNodes-1 number of secondaries (most likely 2) either from available nodes or backups
+   */
+  async select () {
+    // Reset decision tree and backups
+    this.decisionTree = []
+    this.clearBackups()
+
+    // Get all the creator node endpoints on chain and filter
+    let services = await this.getServices()
+    this.decisionTree.push({ stage: DECISION_TREE_STATE.GET_ALL_SERVICES, val: services })
+
+    if (this.whitelist) { services = this.filterToWhitelist(services) }
+    this.decisionTree.push({ stage: DECISION_TREE_STATE.FILTER_TO_WHITELIST, val: services })
+
+    if (this.blacklist) { services = this.filterFromBlacklist(services) }
+    this.decisionTree.push({ stage: DECISION_TREE_STATE.FILTER_FROM_BLACKLIST, val: services })
+
+    const successfulSyncCheckServices = []
```

Comment:
I just realized this logic here replaces the base class `select` logic, which has the concept of "rounds" (e.g. max # of services to try at the same time). This was to protect us from sending too many requests when looking for something like a discprov. I don't think this is a blocker here, but as we expand to > 4 content nodes, we're definitely going to need that logic again.

Reply:
discussed offline -- will implement in future!